cpp/ql/lib/experimental/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -406,7 +425,7 @@ class SsaPhiNode extends Node, TSsaPhiNode {
 
   override Declaration getFunction() { result = phi.getBasicBlock().getEnclosingFunction() }
 
-  override IRType getType() { result instanceof IRVoidType }
+  override DataFlowType getType() { result = this.getAnInput().getType() }
```

Comment:
Was the IRType here wrong? Or has `getType` a different meaning now?

Reply:
The `IRType` here was very wrong, yes. We want the type of a phi node to properly reflect the values that can flow through here, and in that case, that's any of its possible inputs.

The reason this was `IRVoidType` is that I actually thought that our own queries never looked at the type of a dataflow node (but instead did something like `node.asExpr().getType()`), and thus I didn't really bother implementing it properly. However, I then spotted a few queries (for example [this one](https://github.com/github/codeql/blob/main/cpp/ql/src/Likely%20Bugs/Format/NonConstantFormat.ql#L113)) that actually use it. So it should have a proper implementation.